code/game/objects/effects/spawners/windowspawner.dm

```diff
@@ -25,8 +27,14 @@
 				continue
 			var/obj/structure/window/WI = new window_to_spawn_regular(get_turf(src))
 			WI.dir = cdir
+			if(istype(src, /obj/effect/spawner/window/reinforced/polarized) && id)
```

Comment:
Instead of doing this, add a proc `synchronize_variables(var/s)` which you call (instead of using these istypes) with the spawned window as an argument, and define an override on the polarized spawner which sets the ID

Reply:
You are also able to move `id` to a definition on the polarized spawner as well, avoiding it being defined for objects where it's not meaningful.

Reply:
> 
> 
> You are also able to move `id` to a definition on the polarized spawner as well, avoiding it being defined for objects where it's not meaningful.

I was considering this initially, but would have to duplicate a lot of code for an override because of the weird way the windows are spawned in this proc.